rules/type-declaration/src/Rector/Identical/FlipTypeControlToUseExclusiveTypeRector.php

```diff
@@ -0,0 +1,186 @@
+use Rector\Core\Rector\AbstractRector;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\TypeDeclaration\Tests\Rector\Identical\FlipTypeControlToUseExclusiveTypeRector\FlipTypeControlToUseExclusiveTypeRectorTest
+ */
+final class FlipTypeControlToUseExclusiveTypeRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Flip type control to use exclusive type',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function __construct(array $values)
+    {
+        /** @var PhpDocInfo|null $phpDocInfo */
+        $phpDocInfo = $functionLike->getAttribute(AttributeKey::PHP_DOC_INFO);
+        if ($phpDocInfo === null) {
+            return;
+        }
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function __construct(array $values)
+    {
+        $phpDocInfo = $functionLike->getAttribute(AttributeKey::PHP_DOC_INFO);
+        if (! $phpDocInfo instanceof PhpDocInfo) {
+            return;
+        }
+    }
+}
+CODE_SAMPLE
+                ),
+
+            ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Identical::class];
+    }
+
+    /**
+     * @param Identical $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isNull($node->left) && ! $this->isNull($node->right)) {
+            return null;
+        }
+
+        $variable = $this->isNull($node->left)
+            ? $node->right
+            : $node->left;
+
+        $assign = $this->getVariableAssign($node, $variable);
+        if (! $assign instanceof Assign) {
+            return null;
+        }
+
+        $expression = $assign->getAttribute(AttributeKey::PARENT_NODE);
+        if (! $expression instanceof Expression) {
+            return null;
+        }
+
+        $phpDocInfo = $expression->getAttribute(AttributeKey::PHP_DOC_INFO);
+        if (! $phpDocInfo instanceof PhpDocInfo) {
+            return null;
+        }
+
+        /** @var AttributeAwareIdentifierTypeNode[] $types */
+        $types = $this->getTypes($phpDocInfo);
+        if ($this->skipNotNullOneOf($types)) {
+            return null;
+        }
+
+        return $this->processConvertToExclusiveType($types, $variable, $phpDocInfo);
+    }
+
+    /**
+     * @param AttributeAwareIdentifierTypeNode[] $types
+     */
+    private function processConvertToExclusiveType(array $types, Expr $expr, PhpDocInfo $phpDocInfo): ?BooleanNot
+    {
+        $type = $types[0]->name === 'null'
+            ? $types[1]->name
+            : $types[0]->name;
+
+        if (! class_exists($type) && ! interface_exists($type)) {
+            return null;
+        }
```

Comment:
Instead of these checks, the direct object API should be used.

E.g . `$type instanceof IdentifierTypeNode` and similar.

What is the goal here?

Reply:
to ensure that the type is a FQCN type that exists, I just know about `IdentifierTypeNode`, I updated it.

Reply:
@TomasVotruba it cause failure on string type, eg:

```php
        /** @var string|null $string */
        $string = $this->getString();
        if ($string === null) {
            return;
        }
```

see https://github.com/rectorphp/rector/pull/5214/commits/735ea9e6796955b6c6af50e4ed2f7cd674487a46

Reply:
I reverted https://github.com/rectorphp/rector/pull/5214/commits/aa046dff740b621e881d27e3a51e3a3a3c73a59c as it cause issue on https://github.com/rectorphp/rector/pull/5214/commits/735ea9e6796955b6c6af50e4ed2f7cd674487a46